src/contract_wrappers/modules/checkpoint/erc20_dividend_checkpoint_wrapper.ts

```diff
@@ -0,0 +1,578 @@
+        amount: BigNumber,
+    }
+    
+    interface IWithholdingTaxParams {
+        index_0: string,
+    }
+    
+    interface IGetCheckpointDataParams {
+        checkpointId: BigNumber,
+    }
+    
+    interface IDividendsParams {
+        index_0: BigNumber,
+    }
+    
+    interface IIsExcludedParams {
+        investor: string,
+        dividendIndex: BigNumber,
+    }
+    
+    interface ISetWithholdingParams extends ITxParams {
+        investors: string[],
+        withholding: BigNumber[],
+    }
+    
+    interface IDividendTokensParams {
+        index_0: BigNumber
+    }
+    
+    interface IExcludedParams {
+        index_0: BigNumber
+    }
+
+    interface ISetDefaultExcludedParams extends ITxParams {
+        excluded: string[],
+    }
+    
+    interface IPushDividendPaymentParams extends ITxParams {
+        dividendIndex: BigNumber,
+        start: BigNumber,
+        iterations: BigNumber,
+    }
+    
+    interface IGetDividendProgressParams {
+        dividendIndex: BigNumber
+    }
+    
+    interface ICreateDividendParams extends ITxParams {
+        maturity: BigNumber,
+        expiry: BigNumber,
+        token: string,
+        amount: BigNumber,
+        name: string,
+    }
+    
+    interface ICreateDividendWithCheckpointParams extends ITxParams {
+        maturity: BigNumber,
+        expiry: BigNumber,
+        token: string,
+        amount: BigNumber,
+        checkpointId: BigNumber,
+        name: string,
+    }
+    
+    interface ICreateDividendWithExclusionsParams extends ITxParams {
+        maturity: BigNumber,
+        expiry: BigNumber,
+        token: string,
+        amount: BigNumber,
+        excluded: string[],
+        name: string,
+    }
+
+    interface ICreateDividendWithCheckpointAndExclusionsParams extends ITxParams {
+        maturity: BigNumber,
+        expiry: BigNumber,
+        token: string,
+        amount: BigNumber,
+        checkpointId: BigNumber,
+        excluded: string[],
+        name: string,
+    }
+    
+    interface IReclaimDividendParams extends ITxParams {
+        dividendIndex: BigNumber
+    }
+    
+    interface IWithdrawWithholdingParams extends ITxParams {
+        dividendIndex: BigNumber
+    }
+  
+  /**
+   * This class includes the functionality related to interacting with the ERC20DividendCheckpoint contract.
+   */
+  export class ERC20DividendCheckpointWrapper extends ContractWrapper {
+    public abi: ContractAbi = ERC20DividendCheckpoint.abi;
+    private polymathRegistry: PolymathRegistryWrapper;
+    private erc20DividendCheckpointContract: Promise<ERC20DividendCheckpointContract>;
+  
+    /**
+     * Instantiate ERC20DividendCheckpointWrapper
+     * @param web3Wrapper Web3Wrapper instance to use
+     * @param polymathRegistry The PolymathRegistryWrapper instance contract
+     */
+    constructor(web3Wrapper: Web3Wrapper, polymathRegistry: PolymathRegistryWrapper) {
+      super(web3Wrapper);
```

Comment:
Once #36 is merged it should be:
```suggestion
      super(web3Wrapper, address);
```